addons/pkg-npm/common/src/main/java/org/commonjava/indy/pkg/npm/content/DecoratorUtils.java

```diff
@@ -50,7 +51,29 @@ private static String getPath( String url ) throws IOException
         {
             throw new IOException( "Failed to parse URL " + url, e ); // should not happen
         }
-        return url1.getPath();
+
```

Comment:
Why the part 3,2,1 are used for the path? e.g., http://hostname/this/is/a/very/long/sub/path/x.tgz , the String.format( "%s/%s/%s", 5,6,7) would be "sub/path/x.tgz" (the length is 8). The token 1~4 were missing.And this is only for "tarball", would that url happen to have last token "package.json"?